internal/resources/flush_bucket.go

```diff
@@ -0,0 +1,186 @@
+func (c *FlushBucket) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = FlushBucketSchema()
+}
+
+// Create creates a new flush Bucket.
+func (c *FlushBucket) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan providerschema.FlushBucket
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if err := c.validateFlushBucketRequest(plan); err != nil {
+		resp.Diagnostics.AddError(
+			"Error executing bucket flush",
+			"Could not flush the bucket, unexpected error: "+err.Error(),
+		)
+		return
+	}
+	var organizationId = plan.OrganizationId.ValueString()
+	var projectId = plan.ProjectId.ValueString()
+	var clusterId = plan.ClusterId.ValueString()
+	var bucketId = plan.BucketId.ValueString()
+
+	// Execute flush bucket. Nothing gets returned for it.
+	url := fmt.Sprintf("%s/v4/organizations/%s/projects/%s/clusters/%s/buckets/%s/flush", c.HostURL, organizationId, projectId, clusterId, bucketId)
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodPut, SuccessStatus: http.StatusOK}
+	_, err := c.Client.ExecuteWithRetry(
+		ctx,
+		cfg,
+		nil,
+		c.Token,
+		nil,
+	)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error flushing the bucket",
+			errorMessageWhileBucketCreation+api.ParseError(err),
+		)
+		return
+	}
+
+	resp.State.Set(ctx, providerschema.FlushBucket{
+		BucketId:       types.StringValue(bucketId),
+		OrganizationId: types.StringValue(organizationId),
+		ProjectId:      types.StringValue(projectId),
+		ClusterId:      types.StringValue(clusterId),
+	})
+}
+
+// Configure adds the provider configured api to the project resource.
+func (c *FlushBucket) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	data, ok := req.ProviderData.(*providerschema.Data)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *ProviderSourceData, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	c.Data = data
+}
+
+func (c *FlushBucket) Read(_ context.Context, _ resource.ReadRequest, _ *resource.ReadResponse) {
+	// Couchbase Capella's v4 does not support a READ operation for flush endpoint.
+}
+
+func (r *FlushBucket) Delete(_ context.Context, _ resource.DeleteRequest, _ *resource.DeleteResponse) {
+	// Flush endpoint does not support delete as there is nothing on Capella to delete for this resource.
+}
+
+func (c *FlushBucket) ImportState(_ context.Context, _ resource.ImportStateRequest, _ *resource.ImportStateResponse) {
+	// Flush endpoint is not a managed resource on capella. It is purely managed by terraform.
+}
+
+// Flushes the bucket.
+func (c *FlushBucket) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
```

Comment:
Since all the attributes are `requiresReplace`, then `Update` won't be called.

Reply:
Yes, removing it would be good, as having the code is confusing since it doesn't do anything.

Reply:
I've removed it. I left it in just in case, we would want to make changes to flush bucket terraform resources to allow for update too. We could add a simple count of how many times the flush bucket has been executed if we want it be reusable. We can leave it as is, if we are fine with the user having to recreate the resource each time they want to execute a flush on a bucket.

Reply:
> if we are fine with the user having to recreate the resource each time they want to execute a flush on a bucket.

I'm good with this.  Since flush bucket is a destructive operation, the user should be made to explicitly create the resource.